set default shell to bash in shft workflow

RishiKumarRay opened the pull request:
<!-- Thank you for sending a pull request :heart: -->

## Current Behavior

<!-- Please describe the current behavior of the code before the changes in this pull request is applied. -->
Currently the shft workflow doesn't have a default shell option at job level

## Proposed Changes
<!-- Please describe the changes proposed in this pull request. -->
set default shell to bash in publish crates workflow
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [ ] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [ ] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

<!-- Looking forward to merging your contribution!! -->

SanjoDeundiak committed:
chore(rust): remove `ockam_node` message wrapping

SanjoDeundiak committed:
fix(rust): fix `TcpRouter` response address

SanjoDeundiak committed:
feat(rust): add `forward_from_address` function

SanjoDeundiak committed:
feat(rust): fix access controls for api calls

SanjoDeundiak committed:
chore(rust): setup access controls for examples

SanjoDeundiak committed:
feat(rust): deny all for `ockam` crate

SanjoDeundiak committed:
feat(rust): setup access controls for `ockam_api`

SanjoDeundiak committed:
feat(rust): setup access controls for `ockam_channel`

SanjoDeundiak committed:
feat(rust): change `AllowDestinationAddress` access control logic

SanjoDeundiak committed:
feat(rust): add `LocalDestinationOnly` access control

SanjoDeundiak committed:
feat(rust): add convenient contrustor for `DenyAll` mailbox

SanjoDeundiak committed:
feat(rust): add `len` function to `Route`

SanjoDeundiak committed:
feat(rust): set access controls for `ockam_identity`

SanjoDeundiak committed:
feat(rust): improve access control support for `ockam::node` macro

SanjoDeundiak committed:
feat(rust): improve transport access controls

SanjoDeundiak committed:
feat(rust): restrict access control for `send_and_receive`

SanjoDeundiak committed:
feat(rust): restrict access control for `DelayedEvent`

SanjoDeundiak committed:
feat(rust): replace `NodeBuilder` access control to use dynamic dispatch

SanjoDeundiak committed:
feat(rust): use `DenyAll` access control by default for workers and processors

SanjoDeundiak committed:
feat(rust): use `DenyAll` for ble

SanjoDeundiak committed:
feat(rust): setup access controls for `ockam_transport_tcp`

SanjoDeundiak committed:
feat(rust): temporary access controls for `ockam_transport_udp`

SanjoDeundiak committed:
feat(rust): temporary access controls for `ockam_transport_websocket`

SanjoDeundiak committed:
feat(rust): add source and destination access controls

SanjoDeundiak committed:
feat(rust): remove access control support from node macro

SanjoDeundiak committed:
feat(rust): setup access controls for forwarder

SanjoDeundiak committed:
chore(rust): add `TODO` to `Address::random_tagged`

SanjoDeundiak committed:
feat(rust): setup access controls for `ockam_node`

SanjoDeundiak committed:
feat(rust): add generic `Address` to `Mailbox` constructor

SanjoDeundiak committed:
feat(rust): setup access controls for channels and pipes

SanjoDeundiak committed:
chore(rust): add fixme to `ockam_node`

SanjoDeundiak committed:
feat(rust): setup temporary access controls for `ockam_command`

SanjoDeundiak committed:
feat(rust): improve `send_and_receive_with_timeout`

SanjoDeundiak committed:
feat(rust): make access controls a mandatory argument while creating a detached context

SanjoDeundiak committed:
feat(rust): remove unneeded mutability for `Context` instance

SanjoDeundiak committed:
feat(rust): make access controls a mandatory argument while starting a worker or processor

SanjoDeundiak committed:
feat(rust): add `Hop` service to nodes

SanjoDeundiak committed:
test(rust): fix bats tests

SanjoDeundiak committed:
test(rust): change bats tests to pass with access controls updates

SanjoDeundiak committed:
fix(rust): fix `LocalOriginOnly` access control

SanjoDeundiak committed:
feat(rust): improve `RelayMessage`

SanjoDeundiak committed:
feat(rust): improve access controls implementations

SanjoDeundiak committed:
chore(rust): update examples

SanjoDeundiak committed:
docs(rust): update examples in documentation

SanjoDeundiak committed:
feat(rust): rename `LocalOriginOnly` -> `LocalSourceOnly`

SanjoDeundiak committed:
feat(rust): update source and onward access controls and add tests

SanjoDeundiak committed:
feat(rust): avoid using `Arc` on access controls in public api

SanjoDeundiak committed:
test(rust): fix access controls for `ockam_api` identity test`

SanjoDeundiak committed:
feat(rust): further restrict access control for identity secure channel listener

SanjoDeundiak committed:
fix(rust): fix udp example

SanjoDeundiak committed:
fix(rust): chane access control for `ockam_transport_ble` to `AllowAll`

SanjoDeundiak committed:
docs(rust): fix doc comment for `Context`

SanjoDeundiak committed:
feat(rust): remove unused methods from `WorkerBuilder` and `ProcessorBuidler`

SanjoDeundiak committed:
feat(rust): improve `Forwarder` service access controls

SanjoDeundiak committed:
feat(rust): improve `Forwarder` client access control

SanjoDeundiak committed:
chore(rust): remove unwraps

Commit:
chore(rust): add ockam status command

SanjoDeundiak committed:
feat(rust): change `app` `Context` access control to `AllowAll`

SanjoDeundiak committed:
feat(rust): update examples

RishiKumarRay committed:
ci: set default shell to bash in publish crates workflow

RishiKumarRay committed:
Merge branch 'build-trust:develop' into develop

RishiKumarRay committed:
ci:set default shell to bash in Shft workflow

RishiKumarRay committed:
ci:set default shell to bash in Shft workflow

metaclips reviewed (changes requested):
Thanks @RishiKumarRay, can we rebase this PR
```bash
git pull https://github.com/build-trust/ockam.git develop --rebase
```
Also for the current commit message
`ci:set` default shell to bash in `Shft` workflow
Can we change the commit message to 
ci: set default shell to bash in shfmt workflow

We can amend commits with 
```bash
git commit --amend -m "ci: set default shell to bash in shfmt workflow" --no-edit
```

metaclips reviewed (changes requested):
Lint CI seems to fail https://github.com/RishiKumarRay/ockam/actions/runs/3822482430/jobs/6502667627.

SanjoDeundiak committed:
feat(rust): update get started docs, examples and test

SanjoDeundiak committed:
chore(rust): update udp example to use fixes port number

dependabot[bot] committed:
build: bump lru from 0.8.1 to 0.9.0

Bumps [lru](https://github.com/jeromefroe/lru-rs) from 0.8.1 to 0.9.0.
- [Release notes](https://github.com/jeromefroe/lru-rs/releases)
- [Changelog](https://github.com/jeromefroe/lru-rs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jeromefroe/lru-rs/compare/0.8.1...0.9.0)

---
updated-dependencies:
- dependency-name: lru
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

RishiKumarRay committed:
ci: set default shell to bash in shfmt workflow

RishiKumarRay committed:
merged conflict

RishiKumarRay committed:
fix lint issue

RishiKumarRay commented:
@metaclips seems like this PR is messed up , I will open a new PR following all the guidelines
